src/java/org/apache/cassandra/db/RangeTombstoneList.java

```diff
@@ -133,13 +136,21 @@ public void add(ByteBuffer start, ByteBuffer end, long markedAt, int delTime)
         if (c < 0)
         {
             addInternal(size, start, end, markedAt, delTime);
+            assertSorted();
```

Comment:
On the contrary, I completely expect it to throw, if there's already corruption that needs to be hunted down.